examples/robot.py

```diff
@@ -0,0 +1,128 @@
+                              jointUpperLimit=gripper_link_limit[1])
+
+            # Initially set the gripper as opened
+            bc.resetJointState(self.robot_id, self.gripper_link_ids[i],
+                               self.gripper_link_limit[1] * self.gripper_link_multiplier[i])
+
+    def reset_arm_poses(self):
+        for rest_pose, joint_id in zip(self.arm_rest_poses, self.arm_joint_ids):
+            self.bc.resetJointState(self.robot_id, joint_id, rest_pose)
+
+    def open_gripper(self):
+        self.bc.setJointMotorControlArray(self.robot_id, self.gripper_link_ids, self.bc.POSITION_CONTROL,
+                                          targetPositions=[i * self.gripper_link_limit[1] for i in
+                                                           self.gripper_link_multiplier],
+                                          positionGains=[1 for i in range(len(self.gripper_link_ids))])
+
+    def close_gripper(self):
+        # Releasing the position control, leaving a small friction
+        self.bc.setJointMotorControlArray(self.robot_id, self.gripper_link_ids, self.bc.VELOCITY_CONTROL,
+                                          forces=[0.1 for i in range(len(self.gripper_link_ids))])
+        # Force/torque control
+        self.bc.setJointMotorControl2(self.robot_id, self.gripper_link_ids[0], self.bc.TORQUE_CONTROL,
+                                      force=-3 * self.gripper_link_multiplier[0])
+
+    def draw_gripper_frame(self):
+        scale: float = 0.1
+        ee_state = np.array(self.bc.getLinkState(self.robot_id, 6))
+        ee_rotation_matrix = np.array(self.bc.getMatrixFromQuaternion(ee_state[1])).reshape((3, 3))
+        gripper_state = ee_state[0] + ee_rotation_matrix[:, 2] * 0.221
+        self.gripper_x_line = self.bc.addUserDebugLine(lineFromXYZ=gripper_state,
+                                                       lineToXYZ=gripper_state + ee_rotation_matrix[:, 0] * scale,
+                                                       lineColorRGB=[1, 0, 0])
+        self.gripper_y_line = self.bc.addUserDebugLine(lineFromXYZ=gripper_state,
+                                                       lineToXYZ=gripper_state + ee_rotation_matrix[:, 1] * scale,
+                                                       lineColorRGB=[0, 1, 0])
+        self.gripper_z_line = self.bc.addUserDebugLine(lineFromXYZ=gripper_state,
+                                                       lineToXYZ=gripper_state + ee_rotation_matrix[:, 2] * scale,
+                                                       lineColorRGB=[0, 0, 1])
+
+    def delete_gripper_frame(self):
+        if self.gripper_x_line is not None:
+            self.bc.removeUserDebugItem(self.gripper_x_line)
+        if self.gripper_y_line is not None:
+            self.bc.removeUserDebugItem(self.gripper_y_line)
+        if self.gripper_z_line is not None:
+            self.bc.removeUserDebugItem(self.gripper_z_line)
+
+    def get_camera_frame(self):
```

Comment:
This function is a bit ambiguous. I suggest you leave a comment here about your choice of frame convention.
Is this specifying a world_from_camera transform? or camera_from_world transform?

Reply:
Yeah. I tried to get the frame of on-module camera in the robot's wrist in the world frame. Therefore, I changed the function name and added an explanation saying that this is a camera_from_world transform.